torch/distributed/_shard/sharded_tensor/api.py

```diff
@@ -387,7 +388,9 @@ def gather(  # type: ignore[override]
             out (:class `torch.Tensor`, optional): The output full tensor.
                 Must to be provided ONLY on ``dst`` rank.
                 Default: ``None``
-            enforce_dtype: force the imterediate tensor with the same type as input and output
+            enforce_dtype: Deprecated.
```

Comment:
Copying over the comment: can you still keep the enforce_dtype logic `if dtype is None and enfore_dtype is True`? This help us maintain BC, you can throw some warnings in this case to warn user that enforce_dtype is being deprecated